qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -343,6 +351,51 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
         # See: https://github.com/quantumlib/Qualtran/issues/217
         num_toff = 2 * self.a_bitsize * self.b_bitsize - max(self.a_bitsize, self.b_bitsize)
         return {Toffoli(): num_toff}
+    
+    def decompose_from_registers(
```

Comment:
please use the idiomatic way of defining the construction instead of the legacy way... the idiomatic way makes the construction more readable. you can use this as an example

https://github.com/quantumlib/Qualtran/blob/60037bcb9e088d0e22ef436360e6ec262f4cfe88/qualtran/bloqs/arithmetic/comparison.py#L988-L1006